packages/components/inputs/radio-input/src/radio-option.styles.ts

```diff
@@ -133,25 +120,96 @@ const getLabelCursor = (props: TStylesProps) => {
   return 'pointer';
 };
 
-const getLabelStyles = (props: TStylesProps) => css`
-  align-items: center;
-  color: ${getLabelColor(props)};
-  cursor: ${getLabelCursor(props)};
+const RadioOptionKnob = styled.div<TStylesProps>`
+  width: ${designTokens.heightForRadioInputOptionWhenChecked};
+  height: ${designTokens.heightForRadioInputOptionWhenChecked};
+  border-radius: 50%;
+  background-color: ${(props) => getKnobColor(props)};
+`;
+
+const RadioOptionBorder = styled.div<TStylesProps>`
+  width: ${designTokens.heightForRadioInputOption};
+  height: ${designTokens.heightForRadioInputOption};
+  border-radius: 50%;
+  background-color: ${designTokens.backgroundColorForInput};
+  border: ${designTokens.borderForRadioInputOption} solid
+    ${(props) => getBorderColor(props)};
   display: flex;
-  position: relative;
-  &:hover svg *[data-style='radio-option__border'] {
-    stroke: ${getSvgLabelBorderStroke(props)};
+  align-items: center;
+  justify-content: center;
+`;
+
+const Input = styled.input`
+  &:focus + div > ${RadioOptionBorder} {
+    border-color: ${designTokens.borderColorForInputWhenFocused};
+  }
+`;
+
+const getNewThemeHoverStyles = (props: TStylesProps) => {
+  const hoverStyles = css`
+    [class*='getContainerStyles'] {
+      border-color: ${designTokens.colorNeutral90};
+    }
+  `;
+
+  return [
+    !props.isDisabled &&
+      !props.isReadOnly &&
+      /* prettier-ignore */
+      css`
+        &:hover ${hoverStyles};
+      `,
+    props.isHovered && hoverStyles,
+  ];
+};
+
+const getDefaultThemeHoverAndFocusStyles = (props: TStylesProps) => css`
+  &:hover ${RadioOptionBorder} {
+    border-color: ${getLabelBorderColor(props)};
   }
   :focus-within ${LabelTextWrapper} {
     outline: auto 2px ${designTokens.borderColorForInputWhenFocused};
     outline-offset: 3px;
   }
```

Comment:
Please mind this outline in the `focus` state is only added in the current theme (we don't have it in the new theme designs). Not sure if this is expected 
cc @FilPob 

<img width="115" alt="Screenshot 2023-02-21 at 15 21 27" src="https://user-images.githubusercontent.com/49066275/220370841-922ff139-1d11-4294-b274-59e4e42dac22.png">

Reply:
In the CheckboxInput we have it in both. We should do the same here.

<img width="230" alt="image" src="https://user-images.githubusercontent.com/1110551/220430688-c41814f5-1ca9-4611-90a4-42e8364915e5.png">

<img width="251" alt="image" src="https://user-images.githubusercontent.com/1110551/220430601-1ed69037-3f93-462d-b634-cb6181c2c7f4.png">

Reply:
Added here b074705fbfb7916e895d334e776cb1c2d1ac0ce3
I had to decrease the outline for the new theme since it was overlapping with the radio option hover area.
I might add a design token instead if it works better for you.